test/server/doc.test.js

```diff
@@ -0,0 +1,532 @@
+        request
+          .post("/api/documents/")
+          .set("x-access-token", token)
+          .send({
+            title: "New Users Doc From Test",
+            content: "This is a Users doc with a Users role created by the tests",
+            public: true,
+            role: "Users",
+          })
+          .expect(200)
+          .end((err, res) => {
+            expect(res.status).to.exist;
+            expect(res.body).to.exist;
+            expect(res.status).to.equal(201);
+            expect(res.body).to.be.a("object");
+            expect(res.body).to.include.keys("document", "status");
+            expect(res.body).to.have.property("document");
+            expect(res.body).to.have.property("status");
+            expect(res.body.status).to.be.true;
+            expect(res.body.document).to.be.a("object");
+            expect(res.body.document).to.include.keys("_id", "ownerId", "content", "title", "modifiedAt", "createdAt", "public");
+            expect(res.body.document.content).to.be.a("string");
+            expect(res.body.document.title).to.be.a("string");
+            expect(res.body.document.title).to.eql("New Users Doc From Test");
+            expect(res.body.document.public).to.be.true;
+            expect(res.body.document.content).to.eql("This is a Users doc with a Users role created by the tests");
+            done();
+          });
+      });
+      it("should not POST to api/documents without a title", (done) => {
+        request
+          .post("/api/documents/")
+          .set("x-access-token", token)
+          .send({
+            content: "This is will return an error from the db",
+          })
+          .expect(200)
+          .end((err, res) => {
+            expect(res.status).to.exist;
+            expect(res.body).to.exist;
+            expect(res.status).to.equal(400);
+            expect(res.body).to.be.a("object");
+            expect(res.body).to.include.keys("error", "status");
+            expect(res.body).to.have.property("error");
+            expect(res.body).to.have.property("status");
+            expect(res.body.status).to.be.false;
+            expect(res.body.error).to.eql("Title required");
+            done();
+          });
+      });
+
+      it("should not POST to api/documents without a content", (done) => {
+        request
+          .post("/api/documents/")
+          .set("x-access-token", token)
+          .send({
+            title: "This is will return an error from the db",
+          })
+          .expect(200)
+          .end((err, res) => {
+            expect(res.status).to.exist;
+            expect(res.body).to.exist;
+            expect(res.status).to.equal(400);
+            expect(res.body).to.be.a("object");
+            expect(res.body).to.include.keys("error", "status");
+            expect(res.body).to.have.property("error");
+            expect(res.body).to.have.property("status");
```

Comment:
These `expect`s are redundant: if `res.body` has the keys `error` and `status`, then it will also have those properties.